### PR TITLE
[PLT-6631] tests for ButtonIcon component

### PR DESCRIPTION
**Pull request by licarm**
## Description

- outline prop removed from ButtonIcon because it doesn't do anything anymore
- tests in ButtonIcon.test.jsx

### CODE REVIEW

File: packages/riipen-ui/src/components/ButtonIcon.test.jsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .hasClass("focusVisible")
+      ).toEqual(true);
+    });
+
+    it("ddoes not apply focusVisible class when blur event occurs", () => {
```
Comment (imogenem) [inline packages/riipen-ui/src/components/ButtonIcon.test.jsx:223]: spelling error for "does"

File: packages/riipen-ui/src/components/ButtonIcon.test.jsx [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+      expect(
+        wrapper
+          .find("ButtonIcon")
+          .childAt(0)
+          .name()
+      ).toEqual(component);
+    });
+
+    it("sets component with 'button' when component prop is not provided", () => {
```
Comment (imogenem) [inline packages/riipen-ui/src/components/ButtonIcon.test.jsx:102]: Could this be rephrased to something like...

"sets component type to button when component prop is not provided"

**Review by imogenem — CHANGES_REQUESTED** [review]

File: packages/riipen-ui/src/components/ButtonIcon.jsx [hunk 3]
```diff
@@ -58,116 +57,116 @@ const ButtonIcon = props => {
...
-        `}</style>
+        .root {
+          background-color: transparent;
+          border: 0;
+          color: ${theme.palette.text.secondary};
+          cursor: pointer;
+          display: inline-flex;
+          padding: ${theme.spacing(2)}px;
+          position: relative;
+          outline: 0;
```
Comment (jordanell) [inline packages/riipen-ui/src/components/ButtonIcon.jsx:68]: Sick.

File: packages/riipen-ui/src/components/ButtonIcon.test.jsx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+      const errors = jest.spyOn(console, "error").mockImplementation();
+      const component = "abcd";
+
+      mount(<ButtonIcon component={component}>children</ButtonIcon>);
+
+      expect(errors).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe("disabled prop", () => {
```
Comment (jordanell) [inline packages/riipen-ui/src/components/ButtonIcon.test.jsx:136]: I think we are also missing the test "sets disabled on button when disabled prop is true".

**Review by jordanell — CHANGES_REQUESTED** [review]

File: packages/riipen-ui/src/components/ButtonIcon.test.jsx (hunk 4 above)
Comment (licarm) [reply packages/riipen-ui/src/components/ButtonIcon.test.jsx:136]: Is that the same thing as "applies disabled class when disabled prop is true"? It supposed to test line 52 in ButtonIcon.

Comment (jordanell) [reply packages/riipen-ui/src/components/ButtonIcon.test.jsx:136]: Sorry, I don't think I was clear.

    The test you have in this file on line 137 tests line 35 in the source file
    We should add a test to test line 53 in the source file.

**Review by jordanell — APPROVED** [review]

**Pull request merged**